File: Advance/4.Implement a queue using two stacks/Solution.py
class Queue:
    def __init__(self):
        self.s1 = []
        self.s2 = []

    def enqueue(self, value):
        self.s1.append(value)

    def dequeue(self):
        if not self.s2:
            while self.s1:
                self.s2.append(self.s1.pop())
        if not self.s2:
            print("Queue is empty")
            return -1
        return self.s2.pop()

if __name__ == "__main__":
    q = Queue()
    while True:
        print("\n1. Enqueue\n2. Dequeue\n3. Exit")
        choice = int(input("Enter your choice: "))

        if choice == 1:
            value = int(input("Enter value to enqueue: "))
            q.enqueue(value)
        elif choice == 2:
            value = q.dequeue()
            if value != -1:
                print(f"Dequeued value: {value}")
        elif choice == 3:
            print("Exiting...")
            break
        else:
            print("Invalid choice")

File: Advance/4.Implement a queue using two stacks/Solution.java
import java.util.Stack;
import java.util.Scanner;

class Queue {
    Stack<Integer> s1 = new Stack<>();
    Stack<Integer> s2 = new Stack<>();

    void enqueue(int value) {
        s1.push(value);
    }

    int dequeue() {
        if (s2.isEmpty()) {
            while (!s1.isEmpty()) {
                s2.push(s1.pop());
            }
        }
        if (s2.isEmpty()) {
            System.out.println("Queue is empty");
            return -1;
        }
        return s2.pop();
    }
}

public class QueueUsingTwoStacks {
    public static void main(String[] args) {
        Queue q = new Queue();
        Scanner scanner = new Scanner(System.in);
        int choice, value;

        do {
            System.out.println("\n1. Enqueue\n2. Dequeue\n3. Exit\nEnter your choice: ");
            choice = scanner.nextInt();

            switch (choice) {
                case 1:
                    System.out.print("Enter value to enqueue: ");
                    value = scanner.nextInt();
                    q.enqueue(value);
                    break;
                case 2:
                    value = q.dequeue();
                    if (value != -1)
                        System.out.println("Dequeued value: " + value);
                    break;
                case 3:
                    System.out.println("Exiting...");
                    break;
                default:
                    System.out.println("Invalid choice");
            }
        } while (choice != 3);

        scanner.close();
    }
}
